### PR TITLE
A remake of ryandam's Mod for an easy centered cable management

### PR DESCRIPTION
**Pull request by davidauvergne**
* [x] The mod, firmware configuration or slicer profile is in the correct category
    folder. Printable mods go to `printer_mods/`, firmware configurations
    go to `firmware_configurations/`, slicer profiles go to `slicer_profiles/`.
    Create a subfolder with your name, and place the mods in a subfolder with
    a descriptive name within that folder, e.g.: `/printer_mods/FHeilmann/flux_capacitor`
  * [x] Folders names MUST NOT contain spaces. If possible, make sure file names also 
    do not contain any spaces.
  * [x] For each mod, add a small `README.md` file to its folder with a short description
    of what the mod accomplishes. This readme can be used to add pictures, give assembly
    instructions or specify a bill of materials if the mod requires additional hardware.
  * [x] The PR modifies the top-level `README.md` of the category folder adding the 
    contribution to the table. Read the top part of the file for instructions on how
    to do this. Please preserve the alphabetical ordering while adding new rows. Make sure
    to fill out the compatibility matrix to indicate which versions of the Voron printer
    the submission is compatible with.
  * [x] The mod/configuration/profile has been tested by the person submitting the mod 
    and/or other Voron users. Make sure to add information about how the mod was tested below. 
  * [x] The mod is not merely a slight modification of an official Voron part, configuration 
    or profile (i.e. an official Voron part with a few mm added or removed or a slicer profile 
    which only modifies a few values). *(When in doubt, contact one of the admins in the 
    Voron discord before submitting the PR)*
  * [x] Submitted STLs are printable without support. *(If the mod does not meet this criterion
    join the Voron discord and ask the other users for advice on how to modify the mod such 
    that it does not require supports)*
  * [x] Submitted STL files are not corrupt. *(This can be tested by opening the STL in PrusaSlicer
    and checking if mesh errors are reported.)*
  * [x] Submitted STL files are oriented and scaled properly for printing.
  * [x] Submitted firmware configs or slicer profiles contain no sensitive data (e.g. API keys).

#### Which mods/configurations/profiles are added by this PR?
Another Wire Duct for electronic compartment of a V2.4
#### How was it tested? 
Yes
#### Any background context you want to provide?
no
#### Screenshots (if appropriate)
yes
#### Further notes
Thanks :)

### CODE REVIEW

**Comment by davidauvergne** [comment]
A little mod i made and use for all my builds for V2.4

File: printer_mods/davidauvergne/Center_cable_management_duct [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (cbc02009) [inline printer_mods/davidauvergne/Center_cable_management_duct:1]: delete this empty file

File: printer_mods/davidauvergne/README.txt [hunk 2]
```diff
@@ -0,0 +1,2 @@
+A remake of ryandam's Mod for a centered cable management.
```
Comment (cbc02009) [inline printer_mods/davidauvergne/README.txt:1]: can you add a link to the original mod?

**Review by cbc02009 — CHANGES_REQUESTED** [review]
Please add STEP files for source (you can leave the f3d files in too)

File: printer_mods/davidauvergne/README.txt [hunk 3]
```diff
@@ -0,0 +1,5 @@
+A remake of ryandam's Mod for a centered cable management.
+You can find the original here:
+https://github.com/VoronDesign/VoronUsers/tree/master/printer_mods/ryandam/Cable_management_duct
```
Comment (cbc02009) [inline printer_mods/davidauvergne/README.txt:3]: please make this a relative link (`../../ryandam/Cable_management_duct`)

**Review by cbc02009 — CHANGES_REQUESTED** [review]
Please add this mod to the base printer_mods/readme.md so people can find it :)

**Comment by cbc02009** [comment]
Also please double-check the orientation of all the STLs

image

File: printer_mods/davidauvergne/Center_cable_management_duct (hunk 1 above)
Comment (davidauvergne) [reply printer_mods/davidauvergne/Center_cable_management_duct]: Hello, file deleted.

Comment (davidauvergne) [reply printer_mods/davidauvergne/Center_cable_management_duct:1]: And add the mod to the base printer_mods/readme.md

File: printer_mods/README.md [hunk 4]
```diff
@@ -351,3 +351,4 @@ like so:
 | | [V0 Skirt set](./Zen3D/V0_Skirt)| SSR, PowerSwitch, Adxl, Nameplate and color change printing  | :heavy_check_mark: |:x: |:x: |:x:|:x: |
 | | [V0 1515 T-nut](./Zen3D/V0_1515_T-nut)| M3 square and hex nuts for Voron0 1515 frame | :heavy_check_mark: |:x: |:x: |:x:|
 | zruncho | [V0 No-Drop Nuts](./zruncho/V0_No_Drop_Nuts) | Voron0 nut inserts which auto-align (LDO frames only) | :heavy_check_mark: | :x: | :x: | :x: |
+| davidauvergne | [Center_cable_management_duct](./davidauvergne/Center_cable_management_duct) | Mod for a centered cable management | :x: | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
```
Comment (cbc02009) [inline printer_mods/README.md:354]: please add this in alphabetical order.

File: printer_mods/davidauvergne/README.txt [hunk 5]
```diff
@@ -1,5 +1,9 @@
 A remake of ryandam's Mod for a centered cable management.
-You can find the original here:
-https://github.com/VoronDesign/VoronUsers/tree/master/printer_mods/ryandam/Cable_management_duct
+
+You can find the original file here:
+(https://github.com/VoronDesign/VoronUsers/tree/master/printer_mods/ryandam/Cable_management_duct)
```
Comment (cbc02009) [inline printer_mods/davidauvergne/README.txt:4]: Please make this a relative link instead of absolute (../../ryandam/cable_management_duct)

**Review by cbc02009 — CHANGES_REQUESTED** [review]

File: printer_mods/README.md (hunk 4 above)
Comment (davidauvergne) [reply printer_mods/README.md]: Hello, Alphabetical order modified.

File: printer_mods/davidauvergne/README.txt (hunk 5 above)
Comment (davidauvergne) [reply printer_mods/davidauvergne/README.txt]: Relative link is done. Sorry for these errors. Thank you :)

**Review by cbc02009 — APPROVED** [review]

**Comment by cbc02009** [comment]
Everything looks good, but there's a conflict with the main readme.md file that needs to be fixed now. Almost there!

**Comment by cbc02009** [comment]
Hi, you need to pull in the updates from this repo and merge them into your branch to fix the merge conflicts. It's because someone with another 'D' name added a mod while you were working on yours, so git isn't sure how to order the two conflicting commits.

**Comment by davidauvergne** [comment]
> Hi, you need to pull in the updates from this repo and merge them into your branch to fix the merge conflicts. It's because someone with another 'D' name added a mod while you were working on yours, so git isn't sure how to order the two conflicting commits.

Ok, thank you for this information. What do you suggest me to do then ?

**Comment by cbc02009** [comment]
This is because you used the master branch for you mods instead of making a new branch. you can try fetching the upstream by clicking the button in the picture below:
image

github should pop up a text editor that will let you make the changes you need.

**Comment by davidauvergne** [comment]
> This is because you used the master branch for you mods instead of making a new branch. you can try fetching the upstream by clicking the button in the picture below: image
> 
> github should pop up a text editor that will let you make the changes you need.

Hello, I did some modifications, but i'm not sure that it's ok. I'm very sorry if i'm not doing well, i'm beginning to use github.

**Comment by cbc02009** [comment]
I think the easiest thing to do would be:
1. save your mods files on your hard drive somewhere
2. open a command prompt in the clone you're working on
3. run `git reset --hard` to get rid of all your changes and put your master branch back inline with the main master branch
4. click the button in the picture from the last comment (if it's there, if not ignore this)
5. run `git pull` to make your local copy take the updates from the main repo
6. run `git branch cable-raceways` to make a new branch for your changes (replace cable-raceways with whatever you want)
7. put the files you saved in step 1 back into the repo and commit/push them again.
8. open a new pull request, but this time make sure to select the "cable-raceways" branch instead of master. (you might be able to edit this pull request to change the branch, but I don't know how)

Hopefully that makes sense.

**Comment by cbc02009** [comment]
actually, I think I can fix it myself. let me see if this works or breaks something.

**Comment by cbc02009** [comment]
Okay, looks like it worked. Next time you do a mod just make sure to create a new branch for it so that it's not using the master branch :)

**Comment by davidauvergne** [comment]
> Okay, looks like it worked. Next time you do a mod just make sure to create a new branch for it so that it's not using the master branch :)

You are great cbc. Thank you very much for help. Next time i wil work hard to be less NOOB with Github. I promise :)

**Pull request merged**